core/src/main/java/nl/inl/blacklab/index/config/DocIndexerXPath.java

```diff
@@ -517,6 +522,7 @@ protected void processAnnotation(ConfigAnnotation annotation, List<Integer> inde
         for (String captureValuePath: annotation.getCaptureValuePaths()) {
             AutoPilot apCaptureValuePath = acquireAutoPilot(captureValuePath);
             String value = apCaptureValuePath.evalXPathToString();
+            value = processCaptureValuePath(valuePath, i, value, annotation);
```

Comment:
I understand the reason why you need this, but I hope we can arrive at a more generic solution somehow. Let's discuss this next time you're over here. (If we can't think of anything better, I will gladly accept this for the time being)

Reply:
let's do that yes, getting rid of building an xpath using string concatenation is the real problem.

Reply:
The AutoPilot doesn't know of any positional or named parameter construct.

I cannot think of a better way to do this right now.

If we had regular java xpath...XpathVariableResolver would do the trick

Reply:
If I change my setup to this:

```
    - name: lemma
      captureValuePaths:
      - "@xml:id"
      valuePath: "concat(@lemma,following-sibling::tei:join[1]['$1'!='' and contains(@target,'$1')]/@lemma)"
```

there is no need for replacing `, just tested

Reply:
And...tested this again:

```
    - name: lemma
      valuePath: "@lemma"

    - name: lemma
      captureValuePaths:
      - "@xml:id"
      valuePath: "//tei:join[contains(@target,'$1')]/@lemma)"
```
the second definition merges with the first, does not overwrite it.

The processing is also faster.

So....you can never test enough

Reply:
and tested with a condition:

```
    - name: lemma
      captureValuePaths:
      - "@xml:id"
      condition: "@xml:id"
      valuePath: "following-sibling::tei:join[contains(@target,'$1')][1]/@lemma"
```
when the value returned by the xpath in the condition is empty, skip

speeds up indexing by a factor 3